### ISSUE TITLE
`Enum.match()` is unresolved  (and standalone usage fields of Enum too)

### ISSUE THREAD

**Issue by fzzr-** [HaxeFoundation/intellij-haxe#106]
Plugin version: [latest build](https://github.com/TiVo/intellij-haxe/blob/314e4e87dd12e8e2f18893cf105c04b5ef7f382e/intellij-haxe.jar).

![screen shot 2015-01-11 at 18 47 01](https://cloud.githubusercontent.com/assets/888526/5695761/9e4c5cbc-99c2-11e4-8287-cad11037a904.png)

``` haxe
package ;
enum MyEnum { Foo; Bar; }
class Demo {
    public static function main()
    {
        var foo = MyEnum.Foo;
        var bar:MyEnum = Bar; // <- unresolved `Bar`

        // `match` is not resolved:
        MyEnum.Foo.match(MyEnum.Foo);
        Foo.match(MyEnum.Foo); // <- unresolved `Foo`
        foo.match(MyEnum.Bar);
        bar.match(Foo); // <- unresolved `Foo`
    }
}
```

TIR: STB-7321

**Comment by as3boyan** [comment]
Good case.

**Comment by as3boyan** [comment]
Good case.

**Comment by EricBishton** [comment]
Still in 1.3.  I think that the actual EnumValue, while being located properly, is not being treated internally like an EnumValue, so is not checking EnumValue.hx to find its methods.

**Comment by m0rkeulv** [comment]
This could be somewhat solved  in `HaxeEnumModelImpl`, not sure if its the best way to do it, but simply checking the EnumValue model for the member  solves the for me :) 
```
  @Override
  public HaxeMemberModel getMember(@NotNull final String name, @Nullable HaxeGenericResolver resolver) {
    HaxeMemberModel value = getValue(name);
    if (value == null) value = getEnumValueMember(name,  resolver);
    return  value;
  }

  private HaxeMemberModel getEnumValueMember(String name, HaxeGenericResolver resolver) {
    return HaxeEnumUtil.getEnumValueClassModel(this.getBodyPsi()).getMember(name, resolver);
  }
```

**Comment by EricBishton** [comment]
@m0rkeulv The solution in EnumModelImpl works for me.  It would be nice if getHaxeExtendsList and getHaxeImplementsList also just worked, but I don't see an issue with the if statement.  There are probably more things that we have to add in there (like Map variants, etc.) too.

So, do you want to finish this?  Put those tweaks in place and write the unit tests?  😃

**Comment by EricBishton** [comment]
Fixed!  Will be in release 1.4.
